singularity-opac/neutrinos/mean_opacity_neutrinos.hpp

```diff
@@ -0,0 +1,212 @@
+            Real kappaRosselandNum = 0.;
+            Real kappaRosselandDenom = 0.;
+            // Integrate over frequency
+            const int nnu = 100;
+            const Real lnuMin = toLog_(1.e10);
+            const Real lnuMax = toLog_(1.e30);
+            const Real dlnu = (lnuMax - lnuMin) / (nnu - 1);
+            for (int inu = 0; inu < nnu; ++inu) {
+              const Real lnu = lnuMin + inu * dlnu;
+              const Real nu = fromLog_(lnu);
+              kappaPlanckNum +=
+                  opac.AbsorptionCoefficient(rho, T, Ye, type, nu, lambda) /
+                  rho * opac.ThermalDistributionOfTNu(T, type, nu) * nu * dlnu;
+              kappaPlanckDenom +=
+                  opac.ThermalDistributionOfTNu(T, type, nu) * nu * dlnu;
+
+              kappaRosselandNum +=
+                  rho /
+                  opac.AbsorptionCoefficient(rho, T, Ye, type, nu, lambda) *
+                  opac.DThermalDistributionOfTNuDT(T, type, nu) * nu * dlnu;
+              kappaRosselandDenom +=
+                  opac.DThermalDistributionOfTNuDT(T, type, nu) * nu * dlnu;
+            }
+
+            // Trapezoidal rule
+            const Real nu0 = fromLog_(lnuMin);
+            const Real nu1 = fromLog_(lnuMax);
+            kappaPlanckNum -=
+                0.5 * 1. / rho *
+                (opac.AbsorptionCoefficient(rho, T, Ye, type, nu0, lambda) *
+                     opac.ThermalDistributionOfTNu(T, type, nu0) * nu0 +
+                 opac.AbsorptionCoefficient(rho, T, Ye, type, nu1, lambda) *
+                     opac.ThermalDistributionOfTNu(T, type, nu1) * nu1) *
+                dlnu;
+            kappaPlanckDenom -=
+                0.5 *
+                (opac.ThermalDistributionOfTNu(T, type, nu0) * nu0 +
+                 opac.ThermalDistributionOfTNu(T, type, nu1) * nu1) *
+                dlnu;
+            kappaRosselandNum -=
+                0.5 * rho *
+                (1. /
+                     opac.AbsorptionCoefficient(rho, T, Ye, type, nu0, lambda) *
+                     opac.DThermalDistributionOfTNuDT(T, type, nu0) * nu0 +
+                 1. /
+                     opac.AbsorptionCoefficient(rho, T, Ye, type, nu1, lambda) *
+                     opac.DThermalDistributionOfTNuDT(T, type, nu1) * nu1) *
+                dlnu;
+            kappaRosselandDenom -=
+                0.5 *
+                (opac.DThermalDistributionOfTNuDT(T, type, nu0) * nu0 +
+                 opac.DThermalDistributionOfTNuDT(T, type, nu1) * nu1) *
+                dlnu;
+
+            Real lkappaPlanck = toLog_(kappaPlanckNum / kappaPlanckDenom);
+            Real lkappaRosseland =
+                toLog_(1. / (kappaRosselandNum / kappaRosselandDenom));
+            lkappaPlanck_(iRho, iT, iYe, idx) = lkappaPlanck;
+            lkappaRosseland_(iRho, iT, iYe, idx) = lkappaRosseland;
+          }
+        }
+      }
+    }
+  }
+
+#ifdef SPINER_USE_HDF
+  SpinerOpacity(const std::string &filename)
```

Comment:
This should be `MeanOpacity`. Might be worth testing this capability works.

Reply:
Nice catch! Cmake not currently working with HDF5 on for me unfortunately. Should I upgrade the CI to test with HDF5?

Reply:
Worth testing it, I think.

Reply:
CI upgraded